### PR TITLE
Clean up JMX notification listeners for JVM GC metrics

### PR DESCRIPTION
**Pull request by izeye**
This PR changes to clean up JMX notification listeners for JVM GC metrics after binding meters similar to `TomcatMetrics`.

Closes gh-1349

### CODE REVIEW

**Comment by izeye** [comment]
For reviews, see https://github.com/micrometer-metrics/micrometer/pull/1352/files?w=1

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java [hunk 1]
```diff
@@ -133,64 +136,74 @@ public void bindTo(MeterRegistry registry) {
...
 
-                            if (youngGenPoolName != null) {
-                                final long youngBefore = before.get(youngGenPoolName).getUsed();
-                                final long youngAfter = after.get(youngGenPoolName).getUsed();
-                                final long delta = youngBefore - youngGenSizeAfter.get();
-                                youngGenSizeAfter.set(youngAfter);
-                                if (delta > 0L) {
-                                    allocatedBytes.increment(delta);
+                                try {
+                                    ((NotificationEmitter) mbean).removeNotificationListener(this);
```
Comment (mweirauch) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: Maybe my diff viewing skills are fooling me, but isn't this removing the listener exactly after the first notification?

Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: @mweirauch I thought the first notification is enough for binding meters similar to Tomcat one but didn't check. Are the subsequent events considered to get meters bound fully?

Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: @mweirauch Oops. Looking into the code closely, this implementation seems completely wrong. Thanks for pointing this out.

Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: Unless there's a proper shutdown hook, I'm not sure how to remove the listener at the moment.

Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: One way might be to make `JvmGcMetrics` `AutoCloseable`.

Comment (izeye) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java]: @mweirauch Thanks for the feedback! I changed `JvmGcMetrics` to implement `AutoCloseable` although I'm not sure there's a better fix for this.

**Pull request merged**